File: LeetCodeProblemsGo/LeetCode3010DivideAnArrayIntoSubarraysWithMinimumCostI/LeetCode3010DivideAnArrayIntoSubarraysWithMinimumCostI.go
package LeetCode3010DivideAnArrayIntoSubarraysWithMinimumCostI

import (
	"slices"
)

func minimumCost(nums []int) int {
	firstNum := nums[0]
	slices.Sort(nums)
	if nums[2] < firstNum {
		nums[2] = firstNum
	}
	return nums[0] + nums[1] + nums[2]
}

File: LeetCodeProblemsCMake/Problems/LeetCode3010DivideAnArrayIntoSubarraysWithMinimumCostI.cpp
//
// Created by Saeed Jannati on 2/1/2026 A.
//

#include "LeetCode3010DivideAnArrayIntoSubarraysWithMinimumCostI.h"

#include <algorithm>
using namespace std;

int LeetCode3010DivideAnArrayIntoSubarraysWithMinimumCostI::minimumCost(vector<int> &nums) {
    int firstNum{nums[0]};
    ranges::sort(nums);
    if (nums[2]<firstNum) {
        nums[2]=firstNum;
    }
    return nums[0]+nums[1]+nums[2];
}
